opm/models/blackoil/blackoilpolymerparams.cpp

```diff
@@ -0,0 +1,398 @@
+// -*- mode: C++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*-
+// vi: set et ts=4 sw=4 sts=4:
+/*
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+OPM is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+
+Consult the COPYING file in the top-level source directory of this
+module for the precise wording of the license and the list of
+copyright holders.
+*/
+
+#include <config.h>
+#include <opm/models/blackoil/blackoilpolymerparams.hpp>
+
+#if HAVE_ECL_INPUT
+#include <opm/input/eclipse/EclipseState/EclipseState.hpp>
+#include <opm/input/eclipse/EclipseState/Tables/PlyadsTable.hpp>
+#include <opm/input/eclipse/EclipseState/Tables/PlymaxTable.hpp>
+#include <opm/input/eclipse/EclipseState/Tables/PlyrockTable.hpp>
+#include <opm/input/eclipse/EclipseState/Tables/PlyshlogTable.hpp>
+#include <opm/input/eclipse/EclipseState/Tables/PlyviscTable.hpp>
+#endif
+
+#include <cassert>
+#include <cstddef>
+#include <stdexcept>
+#include <type_traits>
+
+namespace {
+std::vector<std::vector<float>> doubleVecsToFloat(const std::vector<std::vector<double>>& input)
```

Comment:
If I'm reading the code correctly, this helper function is unused unless we're building the `float` variants or if we don't `HAVE_ECL_INPUT` (unlikely).  Unused functions in anonymous namespaces may (or may not) generate a warning in some compilers so maybe we could guard the definition by `#if FLOW_INSTANTIATE_FLOAT && HAVE_ECL_INPUT`?

Reply:
indeed. remedied. i also rewrote to use emplace_back.

Reply:
> indeed. remedied. i also rewrote to use emplace_back.

Cool&ndash;thanks. Another option might of course be to make it a template since unused templates are just whitespace:
```C++
template <typename To, typename From>
std::vector<std::vector<To>> convertVecVecTo(const std::vector<std::vector<From>>& src)
{
    std::vector<std::vector<To>> dst{};
    // ...
    return dst;
}
```
and call it as `convertVecVecTo<float>(src)`, but that may just be too much effort for too little gain.

Reply:
if you prefer i can do that. just say which.

Reply:
> if you prefer i can [make it a template]. just say which.

Let's keep it as is.  The template option might become useful if `Scalar` can ever be anything other than `float` or `double`.